src/transaction.rs

```diff
@@ -286,16 +369,64 @@ impl SigningProcess {
     pub fn create_partial_signature(&mut self, wallet: &MultiSig) -> MultiSigResult<PartialSignature> {
         println!();
         println!("2️⃣ Step 2a: Creating your own partial signature.");
-        let data = self.transaction.as_ref().ok_or(MultiSigError::MissingTransaction)?.serialize_content();
         
-        let mut public_keys: Vec<PublicKey> = self.other_commitments.iter().map(|sc| sc.public_key).collect();
-        public_keys.push(self.own_public_key);
-        public_keys.sort();
+        // Aggregate public keys.
+        let aggregated_public_key = self.aggregated_public_key();
+
+        // Hash public keys.
+        let mut public_keys = vec![];
+        for c in self.other_commitments_list.iter(){
+            public_keys.push(c.public_key);
+        }
+        let public_keys_hash = hash_public_keys(&public_keys);
+        // And delinearize them.
+
+        let _delinearized_pk_sum: EdwardsPoint = public_keys
+            .iter()
+            .map(|public_key| public_key.delinearize(&public_keys_hash))
+            .sum();
+
+        let own_kp = KeyPair{
+            public: self.own_public_key.clone(),
+            private: wallet.private_key.clone(),
+        };
+
+        let delinearized_private_key: Scalar = own_kp.delinearize_private_key(&public_keys_hash);
+
+        // Aggregate commitments and hash value b = H(aggregated_public_key|(R_1, ..., R_v)|m).
+        let (aggregated_commitment, b) = self.aggregated_commitment_from_list()?;
+
 
-        let mut commitments: Vec<Commitment> = self.other_commitments.iter().map(|sc| sc.commitment).collect();
-        commitments.push(*self.own_commitment.commitment());
+        // Compute H(commitment || public key || message).
+        let mut hasher = Sha512::new();
+        hasher.update(aggregated_public_key.as_bytes());
+        hasher.update(aggregated_commitment.to_bytes());
+    
+        let data = self.transaction.as_ref().ok_or(MultiSigError::MissingTransaction)?.serialize_content();
+        hasher.update(data);
+
+        let hash = hasher.finalize();
+        let s = Scalar::from_bytes_mod_order_wide(&hash.into());
+        // println!("{:?}", b);
+
+
+        // Compute partial signatures
+        let mut secret = (*self.own_commitment_list[0].random_secret()).0;
+        let mut i = 1;
+        while i < MUSIG2_PARAMETER_V {
+            let mut j = 0;
+            let mut scale = b;
+            while j < i{
```

Comment:
Formatting improvements
```suggestion
            while j < i {
```

Reply:
Actually, a for loop might be better again:
```suggestion
            for j in 0..i {
```